prow/cmd/hook/main.go

```diff
@@ -42,29 +43,59 @@ import (
 	"k8s.io/test-infra/prow/slack"
 )
 
-var (
-	port = flag.Int("port", 8888, "Port to listen on.")
+type options struct {
+	port int
 
-	configPath   = flag.String("config-path", "/etc/config/config", "Path to config.yaml.")
-	cluster      = flag.String("cluster", "", "Path to kube.Cluster YAML file. If empty, uses the local cluster.")
-	pluginConfig = flag.String("plugin-config", "/etc/plugins/plugins", "Path to plugin config file.")
+	configPath   string
+	cluster      string
+	pluginConfig string
 
-	dryRun  = flag.Bool("dry-run", true, "Dry run for testing. Uses API tokens but does not mutate.")
-	deckURL = flag.String("deck-url", "", "Deck URL for read-only access to the cluster.")
+	dryRun  bool
+	deckURL string
 
-	githubEndpoint  = flag.String("github-endpoint", "https://api.github.com", "GitHub's API endpoint.")
-	githubTokenFile = flag.String("github-token-file", "/etc/github/oauth", "Path to the file containing the GitHub OAuth secret.")
+	githubEndpoint  string
+	githubTokenFile string
 
-	webhookSecretFile = flag.String("hmac-secret-file", "/etc/webhook/hmac", "Path to the file containing the GitHub HMAC secret.")
-	slackTokenFile    = flag.String("slack-token-file", "", "Path to the file containing the Slack token to use.")
-)
+	webhookSecretFile string
+	slackTokenFile    string
+}
 
-func main() {
+func (o *options) Validate() error {
```

Comment:
note: didn't validate things that should be impossible given the fact that flags have defaults. unless we think people are going to put together `options` structs by hand that should be fine?

Reply:
since it's internal to main() I think that's not fine [constructing by hand]